examples/2-advanced/variable-depth.ipynb

```diff
@@ -0,0 +1,521 @@
cwZIIxALBBEIJjXXP7/I6J6a3tcgijWPEEjhlKbJxD51wvFGCfUNU1IZaPdhBgyy+Jc7bjx6pbFSfsWaY8hhMAksfiSp7SSHZLK0lxlZJKNtq8kTlMOF1rKptmLyH7O40K7dxLbN7AEz2fioBrTkenhxtdmqXwxP3fEP7PzAUACggioyYUQUToxRORe16ONm2Kz1VDzBOkPJ9SQQSOGuHZ8X132yhfXfV4cIjqmO1Zwl2Rjm6x47do0EwGgRba5dPvKudTZoaYzQtp9jKSxQ/cpcpWCufZGij1uCsk8wbWHkaYEL8Y4oTwvd5mdPYb0nLiYNhBGQAsEEXBiiiATTcYmlYGCSwxp1gyFZpfUZguaNoGCSCOGum6aEHvdFz+UUIMGFxBDYJKxRZFGEGld5doQQz40GSFNVsUWRcPzZdtv+3HX0IislNkkl/V2nexRbHzu55UDcQRcQBABlpMu/1MiCl+DI7dpx0Chq+YJsWVydrtQMZRKCA2vhcUKb+Nt4o2fUvzc8vkDIYTAVLHNpdvTg8/+NRERbfvj7dSiRRINKTdZjV2PxO+X5M8C2fOSsjga5mhA89Rv/dhFaIbKl0Fjxwg0TgjJHpljSKV1sWYQEEaAA4IIlMiFEFEzYihVTN9Gq764dcwTmjZOKI2VUAzVKWObBDGUAgggMEts/+Nt2fOSILIZV3ZIY5AQut6nbtZnjsazvkgjjjRoM0uqWIH7JdUxUQiND2EEJCCIABGVhVBOKrODuoLIW+KWwDxBM49Q84SwNuXHdfYW4uYZK4ZiSuTq7CWkIWU2yARiCMwitihqQwyF7CWkia8tgdNkTYrxrawMEamyNyHkY3PleaHri+apX5qn5phDMx+tVXjK7E6oMYMv/hODuSLm+Yf/C/tagNkCgmjGef0Vb1J9CxNrduAyUPDFUK25idjENVQM+YSQFFPXhrxt1PGV/fxjiV0UfXUCqNfTjxUaPwaIITDLmKKoTUFkx9PsIcTH1gmR0CyQJHBSGitoaNKsITTDpI3tM22ozCOwzM5nzBBaxkcEYTTrQBDNKK+/4k3Fcawgcmd16q8ZUgmKQPMEO443k+S4Js0rdr2QVmRos0KumNK1NsQQkd4+u6lsUA6EEAAjllz25OK47qarkimBT2CFWHe7YkpZDkmExGR3bFzZldD5cNSJXzf7pJkP36Zb2aPS84IwAgYQRDOITwzVLZXTfDNU10DBN8cUYogojZNcqIucdC6VcYLrWqyLXJOlcU0BMQRAlVwUaQVR/k08J4Zs0mWbZKOEkAyMVO4WYrvNz685a+6UWaLQvpoSQi6uxsCBKD67kzJ79IfBHBFBFM0iEEQzhCmEcrg3C2cb8VsgtyAKzeRIYkgzR9c8JfHiy7rEiqFqG7aJ1zxhnGJoHHbaEEMAjI8dLt9avMZlb2xSCKI65XG+DIwmG1TXZCF1dqgOPnEkzSU0k0SkK8HL44Zmj0rjODJSrjEr55n4fxjMFfP6z2d+iu0Hpg8Iohngj35yOi9SArMubJ/K47BSOY2QCTFQ0IgSdR/HvLpgqw0xpGPFa9fSLZ8/sDgGAPgxRdE9RzxUPM7Fy7ic5LTlcaUxNmUwOHMBrr8mw5RyjVA+R3Nuqey3TXzizPeFpytbZhNj4DD8N132KKbETtocFsJo+oEgmmL+6CenF8eS7aRJnTVD7DgBGSdVdiUig+XLtoSKoVgnubb2GIIYKgMBBEA6dr5icXEcK4gk84RS28DskEsE5cc+YrI1vnFDibXuTmW9bRJbfuciJHtUPhdn0CDFMcWW5n4Iwmg2gCCaQkwhlOMTRCkc5SrjBKzziRFEvuySJovjWjNkX++KrTY3z9S22m2JoSzrUW/TTUWsGOKMGno9iCEAmmDnKxbXyg65zBOqa5H89twakwRfRsg+drXJr7lK3kLtr+tYd9v4Yqey4TYf25/lPiMHzRyktUcpDRpC1h3Z8f7tsH9lXy8wuUAQTRGnXvlG9puR1NkhuQ0vhmLjhdhre/t7+kh/BP7MT5gYqmOrXbT19HOdL7eRzmv6cq93mJ12imyQNM5uJ0EMAdAUu16xFXs+v/mVTA9Slcf5StI0BgmajJArfso9iVJZaIfEk2JzwiikVE+zXskHN5bmS9jSeU+JnSqG8Vo9kf8+W/EgjKYHCKIp4NQr31gcSzaSo3OWEIlYkxO6big0Xoi9dh2zA58gSl0qFypefPOLEUO+v/bYzVZj7bTrADEEwPiwRRFnrW1ndlKYJ8QYJsRmhFyEZIFC5qI55qgTTxJGmuflK4Nzff6G2ICbwohdAxSQPapj7f1E1q/EgyiaDiCIJhhTCBH5xdDwvFyCVrmuuFGvK4bs8ynEkB3HW1rnmRM3L91aJXlO7n7++Y1DDMXGDY3nGseVgYIQAqB9cmFkZoc4QrJDMeVxdmbGlQ3yZY1cGRGNVbd2HA1zvQHNb/qcDT0OxZU98mWrfBkzTZbK10Zr8z06z6wZSrwxrBmvTxl9+rBPSy8R6DgQRBOILYRyxi2I6mabfHsNaeKldpOzY+rGZMaIzGSFOMmNQwxpYofGc41hZ6AgggDoDst+sihIEMU4yVXiKp3jpHghmSZ7HGlMH5os1BzzGsxn/eK8dCwRK5I4Ysr4pAxPSJbIFEaatU/Vc/WtvV1LEMyYEEaTBwTRBHHalacRUfWPN6Ym1ic2dI/1jnKhpXIpxJBvfKL6Yqjaxn09JHaIGJLOa/+6QwVWaHxtvJAxIIYA6B4rrtyyci42O+TLqkiiRTZ3qJojmHF8Bgdc5kJT0ubL5riup4YTUnWyTqFrn1y24lI833h5XCleCmtvV+boiU2buXLzgjCaHCCIJgRODEnWkkRCKZsgOHyZIM1ixhAB4xMWofbamjU4PsERuxYpdNNVbVx7fq5Y8hj8fELnJ/fTxdfGCxkDYgiA7mKKolgjhZB1POX+bjEUEismjhg/QtxoMj8pqZNpkgRhSDYpJNujuc7F09zPFOcCbb3NO
```

Comment:
This section was initially

```python
for i in range(nbins):
    for j in range(len(zbins)):
```

and displayed every single `hp.mollview`. This took a long time and made the notebook huge, so I have just made it one.